<a href="https://colab.research.google.com/github/anjilakarkigit/discreteMathematics/blob/main/Project4_counting.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Project Part 4 : CPSMA 3133 Dr. Jacob's Discrete Mathematics 

## Counting

### Anjila Karki and 267815




# Counting

In [2]:
possibleProtein = ['beef','chicken','pork']

For the counting part of the project, I will be looking at the different protein on a keto diet that is mostly connsumed by an individual.I have three different types of protein option on my sequence of ten.Therefore, there are $3^{10}$ possibilities.

$$ 3*3*3\dots *3 = 3^{10} $$

In [3]:
3**10

59049

So, from the result there are total 59049 number of possibilities.

Now, on the next part we can use pigeanhole principle where n is the number spaces and m is the number of categories.from the above condition we have 10 spaces to fill three categories.

$$
\frac{(n+m-1)!}{n!(m-1)!}
$$

In [5]:
import math

math.factorial(12)/(math.factorial(10)*math.factorial(2))

66.0

So, there is only 66 possibilities if the order didn't matter, and if I compare the the both the possibilities there is a huge difference.

Now, I am going to look at the possible outcomes if not allowed to select an item multiple times. There are two condition which are order matters and order doesn't matters.If not allowed to repeat the categories we cannot fill 10 spaces. We can only have 3 spaces maximum for 3 categories.

Lets look at if the order matter first. The possible outcome if the order matter is $3!$

In [6]:
math.factorial(3)

6

In [8]:
from itertools import permutations
print(list(permutations(possibleProtein,3)))

[('beef', 'chicken', 'pork'), ('beef', 'pork', 'chicken'), ('chicken', 'beef', 'pork'), ('chicken', 'pork', 'beef'), ('pork', 'beef', 'chicken'), ('pork', 'chicken', 'beef')]


So, there are 6 possible outcome if the order matter,Which are listed below the number of outcomes.

Now, for it the order doesn't matter, we use 

$nCm = \frac{3!}{0!.3!}$


In [7]:
math.factorial(3)/(math.factorial(0)*math.factorial(3))

1.0

In [9]:
from itertools import combinations
print(list(combinations(possibleProtein,3)))

[('beef', 'chicken', 'pork')]


So, from the result I can see only one possible outcome, which is listed above.

Now, on the next part, we can calculate the 5th term polynomial $(x-3)^9$ by the formula:

$$
T{r+1} = nCrx^{n-r}y^r
$$

$$
T{4+1} = 9C4x^{5}(-3)^4
$$

In [10]:
(math.factorial(9)/(math.factorial(5)*math.factorial(4)))*((-3)**4)

10206.0

So, the fifth term is $$10206x^5$$